scripts/index_to_yaml.py

```diff
@@ -12,10 +14,63 @@
 from simplesat.constraints.kinds import (
     Any, EnpkgUpstreamMatch, Equal
 )
-from simplesat.test_utils import repository_from_index
+from simplesat.package import (
+    PackageMetadata, RepositoryInfo, RepositoryPackageMetadata
+)
+from simplesat.repository import Repository
+
+
+def parse_index_packages_to_install_requires(packages):
+    name_to_install_requires = {}
+
+    for dependency in packages:
+        parts = dependency.split(None, 1)
+        name = parts[0].lower()
+        name_to_install_requires.setdefault(name, [])
+
+        if len(parts) == 1:
+            constraint = "*"
+        else:
+            version_part = parts[1]
+            package_version = EnpkgVersion.from_string(version_part)
+            if package_version.build == 0:
+                constraint = "^= {0}".format(version_part)
+            else:
+                constraint = "== {0}".format(version_part)
+
+        name_to_install_requires[name].append(constraint)
+
+    return dict(
+        (name, tuple(value))
+        for name, value in six.iteritems(name_to_install_requires)
```

Comment:
You iterate over the entries to create a `dict` and then immediately iterate over the `dict` to create a `tuple`. Why not just create the form that you want here?